patroni/ha.py

```diff
@@ -455,6 +451,22 @@ def recover(self) -> str:
 
         data = self.state_handler.controldata()
         logger.info('pg_controldata:\n%s\n', '\n'.join('  {0}: {1}'.format(k, v) for k, v in data.items()))
+
+        # timeout > 0 indicates that we still have the leader lock and it was just updated
```

Comment:
This method is quite complex, it could do with being broken down into smaller methods.

pylint
```
patroni/ha.py:438:4: R0912: Too many branches (17/12) (too-many-branches)
```

Also, it doesn't have a docstring.

Reply:
hehe, been there. It is feature, not a bug  🥲 

https://twitter.com/hugh_capet/status/1592487213274918912?s=46&t=EUhTKJup0nNtK1f7R-fEgA